receiver/sqlserverreceiver/scraper.go

```diff
@@ -573,110 +575,143 @@ func (s *sqlServerScraperHelper) recordDatabaseQueryTextAndPlan(ctx context.Cont
-		record.Attributes().PutStr(dbPrefix+queryPlanHash, queryPlanHashVal)
 
-		s.logger.Debug(fmt.Sprintf("QueryHash: %v, PlanHash: %v, DataRow: %v", queryHashVal, queryPlanHashVal, row))
-
-		record.Attributes().PutDouble(dbPrefix+totalElapsedTime, float64(totalElapsedTimeDiffs[i])/1000)
-
-		// handling `total_rows`
-		rowsReturnVal, err := strconv.ParseInt(row[rowsReturned], 10, 64)
-		if err != nil {
-			err = fmt.Errorf("row %d: %w", i, err)
-			errs = append(errs, err)
-		}
-		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, rowsReturned, rowsReturnVal); cached {
-			record.Attributes().PutInt(dbPrefix+rowsReturned, diff)
-		}
-
-		// handling `total_logical_reads`
-		logicalReadsVal, err := strconv.ParseInt(row[logicalReads], 10, 64)
-		if err != nil {
-			err = fmt.Errorf("row %d: %w", i, err)
-			errs = append(errs, err)
-		}
-		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, logicalReads, logicalReadsVal); cached {
-			record.Attributes().PutInt(dbPrefix+logicalReads, diff)
+		attributes := []internalAttribute{
+			{
+				key:            computerNameKey,
+				columnName:     computerNameKey,
+				valueRetriever: vanillaRetriever,
+				valueSetter:    setString,
+			},
+			{
+				key:        "db.query.text",
+				columnName: queryText,
+				valueRetriever: func(row sqlquery.StringMap, columnName string) (any, error) {
+					return obfuscateSQL(row[columnName])
+				},
+				valueSetter: setString,
+			},
+			{
+				key:            dbPrefix + executionCount,
+				columnName:     executionCount,
+				valueRetriever: retrieveInt,
+				valueSetter:    setInt,
+			},
+			{
+				key:            dbPrefix + logicalReads,
+				columnName:     logicalReads,
+				valueRetriever: retrieveInt,
+				valueSetter:    setInt,
+			},
+			{
+				key:            dbPrefix + logicalWrites,
+				columnName:     logicalWrites,
+				valueRetriever: retrieveInt,
+				valueSetter:    setInt,
+			},
+			{
+				key:            dbPrefix + physicalReads,
+				columnName:     physicalReads,
+				valueRetriever: retrieveInt,
+				valueSetter:    setInt,
+			},
+			{
+				key:            dbPrefix + queryHash,
+				valueRetriever: defaultValueRetriever(queryHashVal),
+				valueSetter:    setString,
+			},
+			{
+				key:        dbPrefix + queryPlan,
+				columnName: queryPlan,
+				valueRetriever: func(row sqlquery.StringMap, columnName string) (any, error) {
+					return obfuscateXMLPlan(row[columnName])
+				},
+				valueSetter: setString,
+			},
+			{
+				key:            dbPrefix + queryPlanHash,
+				valueRetriever: defaultValueRetriever(queryPlanHashVal),
+				valueSetter:    setString,
+			},
+			{
+				key:            dbPrefix + rowsReturned,
+				columnName:     rowsReturned,
+				valueRetriever: retrieveInt,
+				valueSetter:    setInt,
+			},
+			{
+				key:            dbPrefix + totalElapsedTime,
+				valueRetriever: defaultValueRetriever(float64(totalElapsedTimeDiffsMicroSecond[i]) / 1000 / 1000),
```

Comment:
```suggestion
				valueRetriever: defaultValueRetriever(float64(totalElapsedTimeDiffsMicroSecond[i]) / 1000_000),
```

You could use `1000_000` to replace `1000 / 1000`.